internal/core/app_status_report.go

```diff
@@ -0,0 +1,340 @@
+package core
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/zclconf/go-cty/cty"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/go-argmapper"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl/v2"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/component"
+	sdk "github.com/hashicorp/waypoint-plugin-sdk/proto/gen"
+	"github.com/hashicorp/waypoint/internal/config"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+func (a *App) DeploymentStatusReport(
+	ctx context.Context,
+	deployTarget *pb.Deployment,
+) (*pb.StatusReport, *sdk.StatusReport, error) {
+	var evalCtx hcl.EvalContext
+	// Load the deployment variables context
+	if err := a.deployStatusReportEvalContext(ctx, deployTarget, &evalCtx); err != nil {
+		return nil, nil, err
+	}
+
+	// Load variables from deploy
+	hclCtx := evalCtx.NewChild()
+	if _, err := a.deployEvalContext(ctx, hclCtx); err != nil {
+		return nil, nil, err
+	}
+
+	c, err := a.createStatusReporter(ctx, hclCtx, component.PlatformType)
+	if status.Code(err) == codes.Unimplemented {
+		c = nil
+		err = nil
+	}
+	if err != nil {
+		a.logger.Error("error creating component in platform", "error", err)
+		return nil, nil, err
+	}
+
+	a.logger.Debug("starting status report operation")
+	statusReporter, ok := c.Value.(component.Status)
```

Comment:
`c` can be nil here from the Unimplemented block, which will crash I believe?

Reply:
From the testing now, it looks like you're right in that even if Status is unimplemented, the component creator still returns something and doesn't raise an error. We do get a component back, so this conversion doesn't crash. I'll still clean this up to handle this scenario properly where `c` ends up nil

Reply:
I resolved this for now by returning early when we get an Unimplemented error so that should prevent this here.